src/FakeItEasy/Core/AssemblyExtensions.cs

```diff
@@ -14,7 +14,16 @@ public static bool ReferencesFakeItEasy(this Assembly assembly)
         {
             Guard.AgainstNull(assembly, nameof(assembly));
 
+#if FEATURE_REFLECTION_GETASSEMBLIES
             return assembly.GetReferencedAssemblies().Any(r => r.FullName == TypeCatalogue.FakeItEasyAssembly.FullName);
+#else
+            var fakeItEasyLibraryName = TypeCatalogue.FakeItEasyAssembly.GetName().Name;
+            var context = Microsoft.Extensions.DependencyModel.DependencyContext.Default;
+
+            var runtimeLib = context.RuntimeLibraries
+                .SingleOrDefault(library => string.Equals(library.Name, assembly.Name(), System.StringComparison.Ordinal));

```

Comment:
I think DependencyContext provides compilation  information at runtime. If you look under the dotnet build output folder you should see an  <assembly>.deps.json. As far as I can see under  debugger, `context` contains information from  that file.  `CompileLibraries` include all the non-meta libraries in `targets` section, while `RuntimeLibraries` seem to include all the runtime dependencies.